### DISCUSSION TITLE
Reporting tool

### DISCUSSION THREAD

**Discussion by kcabral817** [MarimerLLC/csla#3665] [Questions]
Hello, 
I am looking for a reporting tool that will work on Blazor and CSLA.  Some of the reporting tools I've looked at don't use the DI container, don't use business objects as the source, so I was wondering what others are using.  Since this is a SaaS Azure hosted solution, access to the database by anything other than CSLA is verboten.  Any suggestions would be greatly appreciated.

**Comment by danielmartind** [comment]
Have you looked at Bold Reports?

**Comment by kcabral817** [comment]
I'm reviewing right now, but I haven't gotten it work yet with a business object.  Have you gotten to work in your organization?
________________________________

**Reply by danielmartind** [reply]
Nothing in production but have got in working in WPF and hosting a WPF control in Winforms using CSLA 7. I'm using RDLC reports and was looking for a way to migrate to current versions of .NET from 4.8.

**Reply by Chicagoan2016** [reply]
@danielmartind  We have been using rdlc reports, we are in .NET 4.7 (and 4.8). Do rdlc reports not work in current versions of .NET (.NET 8.0)?

**Reply by danielmartind** [reply]
ReportViewer was not migrated to the new .NET. I think they want you to use their BI platform. Bold Reports has a control for most UI apart from Winforms but you can embed the WPF version. It  lets you use your RDL/RDLC reports. There is a paid license or free cuminity version for small businesses. I've only tested that I can make it work so not sure how it matches up to report viewer.

**Comment by JacoJordaan** [comment]
You can also look at Telerik reporting

**Comment by kcabral817** [comment]
I just opened a support ticket with boldReport to determine the business object connection issue.  I'm using blazor server .net with csla.  I looked at Telerik, but their business object solution actually uses an excel in their demo, so I am not sure if it works with business objects.

Have a nice day.

Kevin

https://rb.gy/syiuf


________________________________

**Comment by kcabral817** [comment]
my difficulty is with the binding.  I haven't been able to get data to be displayed in an RDLC report.  The report shows up sans data.

Have a nice day.

Kevin

https://rb.gy/syiuf


________________________________

**Reply by danielmartind** [reply]
Looking at the [docs]( https://help.boldreports.com/embedded-reporting/blazor-reporting/report-viewer/rdlc-report/ ) you set the binding in OnInitReportOptions method in your report controller. 

Load your CSLA objects, I always use a ReadOnlyList and add your datasource

``reportOption.ReportModel.DataSources.Add(new BoldReports.Web.ReportDataSource {Name="MyReportDataSourceName", Value=MyReadonlyList});``

Object has to be a list and the name has to match the datasource name in th report. Looks similar to what I did in WPF except I was adding an array or data sources as had serveral in the report.

**Comment by kcabral817** [comment]
In a blazor application I am unable to get it to work.

Here is my assignment in public class BoldReportsAPIController
  reportOption.ReportModel.DataSources.Add(new ReportDataSource { Name = "ProposalReport", Value = ProductList.GetData() });

 Here is the RDLC declaration:

 <rd:DataSetInfo>
   <rd:DataSetName>ReportExplore1</rd:DataSetName>
  <rd:ObjectDataSourceType>DataSource1.ProductList, ReportExplore1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null</rd:ObjectDataSourceType>
  <rd:TableName>ProposalReport</rd:TableName>
 </rd:DataSetInfo>

Kevin

https://rb.gy/syiuf


________________________________

**Comment by MarkOverstreet** [comment]
We have used DevExpress reporting for years.  Try it, it works great for us but it is expensive!

**Reply by amselem** [reply]
Another happy devexpress XtraReports user here, they work fine with objects as datasource
